### PR TITLE
Vscodeconfig

### PR DESCRIPTION
**Pull request by anaida-gasparyan**

### CODE REVIEW

**Review by friedchipz — DISMISSED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+CPP		:= g++
+C_FLAGS := -std=c++17 -Wall -Wextra -O3
+
+ROOT	:= Pong1
+BINDIR	:= $(ROOT)/bin
+LIBDIR	:= $(ROOT)/lib
```
Comment (anaida-gasparyan) [inline Makefile:6]: `LIBDIR ` is not used

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+INCDIR	:= $(ROOT)/include
+SRCDIR	:= $(ROOT)/src
+OBJDIR	:= $(ROOT)/obj
+
+SRC		= $(notdir $(wildcard Pong1/src/*.cpp ))
```
Comment (anaida-gasparyan) [inline Makefile]: `Pong1/src` is actually `SRCDIR`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+OBJ		:= $(addprefix Pong1/obj/, $(patsubst %.cpp, %.o, $(SRC)))
```
Comment (anaida-gasparyan) [inline Makefile]: same for this `Pong1/obj` as `OBJDIR`

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+else
+	EXECUTABLE	:= main
+	SDL_INCLUDE	:= /usr/local/include/SDL2
+	SDL_LIB		:= /usr/local/Cellar/sdl2/2.0.8/lib
+	libsdl		:= $(SDL_LIB)/libSDL2.so
+endif
+
+.PHONY : clean veryclean directories link all
+
+all: directories link
```
Comment (anaida-gasparyan) [inline Makefile]: please note that is `tasks.json` the `make run` is mention to actually run. so better still having `make build` (or some alternative) and `make run` for running and just `make` for both

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+directories:
+	@mkdir -p $(OBJDIR)
+	@mkdir -p $(BINDIR)
+	@mkdir -p $(LIBDIR)
+
+clean:
+	$(RM) -r $(OBJDIR)
+
+veryclean: clean
```
Comment (anaida-gasparyan) [inline Makefile:42]: why do we need separate `veryclean`?

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	$(RM) -r $(BINDIR)
+
+Pong1/obj/%.o : Pong1/src/%.cpp
```
Comment (anaida-gasparyan) [inline Makefile]: Do we want to replace this `Pong1/obj` with `OBJDIR`?

File: Makefile (hunk 2 above)
Comment (friedchipz) [reply Makefile]: yes, but you can't use it inside the wildcar invocation (i may be wrong about this, but i didn't manage to make it work)

File: Makefile (hunk 3 above)
Comment (friedchipz) [reply Makefile]: same as for SCRDIR

File: Makefile (hunk 4 above)
Comment (friedchipz) [reply Makefile]: we will look into it

File: Makefile (hunk 2 above)
Comment (anaida-gasparyan) [reply Makefile]: let me have a quick google search about it ;)

**Review by friedchipz — APPROVED** [review]
good contributions and fixes :D

**Pull request merged**